tower-discover/src/list.rs

```diff
@@ -31,15 +34,19 @@ where
 impl<T, U> Discover for ServiceList<T>
 where
     T: IntoIterator<Item = U>,
+    Enumerate<T::IntoIter>: Unpin,
```

Comment:
Why is Unpin required here?

Reply:
https://github.com/taiki-e/pin-project/issues/76

Reply:
https://github.com/taiki-e/pin-project/issues/76